### PR TITLE
Add readme.md files to each project

### PR DESCRIPTION
**Pull request by peter-csala**
# Description
- I've added a readme.md file to each project
- The markdown files are following the same structure
- I've also added several `mermaid` diagrams to help understand project relationships

# Previews:
- [PollyDemos](https://github.com/peter-csala/Polly-Samples/blob/migrate-readme-md/PollyDemos/README.md)
- [PollyTestWebApi](https://github.com/peter-csala/Polly-Samples/blob/migrate-readme-md/PollyTestWebApi/README.md)
- [PollyTestClientConsole](https://github.com/peter-csala/Polly-Samples/blob/migrate-readme-md/PollyTestClientConsole/README.md)
- [PollyTestClientWpf](https://github.com/peter-csala/Polly-Samples/blob/migrate-readme-md/PollyTestClientWpf/README.md)

### CODE REVIEW

**Comment by peter-csala** [comment]
@martincostello In the next PR I will update the main `readme.md`. I plan to remove the migration note from the main page if you don't mind, since the migration can be considered as done.

File: PollyTestClientConsole/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+- It lets you start and stop any demo.
+- While the demo is running it prompts the logs.
+- To stop the demo press _any_ key.
+- When you stop the demo you can scrutinize the logs and check the statistics.
+- To exit from the application press _any_ key after you assessed the prompted messages.
+
+## Structure
+
+- The `Program.cs` is the main entry point.
```
Comment (martincostello) [inline PollyTestClientConsole/README.md:28]: Let's make `Program.cs` a relative link to the actual code file.

File: PollyTestClientConsole/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+- To run a new demo: comment out the current one and uncomment the next.
```
Comment (martincostello) [inline PollyTestClientConsole/README.md:29]:
```suggestion
- To run a new demo, comment out the current one and uncomment the next.
```

File: PollyTestClientConsole/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+- To switch from the sync to async demos overwrite the `useSyncDemos` to `false`.
```
Comment (martincostello) [inline PollyTestClientConsole/README.md:30]:
```suggestion
- To switch from the sync to async demos change the value of the `useSyncDemos` variable to `false`.
```

File: PollyTestClientConsole/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+- We suggest to run the demos in the same order as they are defined in this file.
```
Comment (martincostello) [inline PollyTestClientConsole/README.md:31]:
```suggestion
- We suggest to run the demos in the same order as they are defined.
```

File: PollyTestClientWpf/README.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+- It provides a drop down list from which you can choose which demo do you want to run.
+- It shows some minimal description about the chosen demo.
+- It lets you start, stop and rerun any demo.
+- While the demo is running
+  - it prompts the logs with auto-scrolling and
+  - it auto-updates the statistics.
+- When you stop the demo
+  - you can scrutinize the logs and
+  - check the final statistics.
```
Comment (martincostello) [inline PollyTestClientWpf/README.md:28]:
```suggestion
- When you stop the demo:
  - you can scrutinize the logs and;
  - check the final statistics.
```

File: PollyTestClientWpf/README.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+- When you start a new demo or restart the previous one
+  - then the logs and statistics are reset.
```
Comment (martincostello) [inline PollyTestClientWpf/README.md:30]:
```suggestion
- When you start a new demo or restart the previous one then the logs and statistics are reset.
```

File: PollyTestClientWpf/README.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+## Structure
+
+- The `MainWindow.xaml` defines the UI components and their layout.
+- The `MainWindow.xaml.cs`
```
Comment (martincostello) [inline PollyTestClientWpf/README.md:35]: Let's make these relative links too.

File: PollyTestClientWpf/README.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  - handles the interactions
+  - uses reflection to invoke the selected demo
+  - updates UI whenever there is a progress change
```
Comment (martincostello) [inline PollyTestClientWpf/README.md:38]:
```suggestion
  - updates the UI whenever there is a progress change
```

File: PollyTestClientWpf/README.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  - handles all sort of errors.
+
+## How to run?
+- From the `PollySamples` directory:
```
Comment (martincostello) [inline PollyTestClientWpf/README.md:42]:
```suggestion
- From the `PollySamples` directory:

```

File: PollyTestClientWpf/README.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+```none
+dotnet run --project PollyTestClientWpf/PollyTestClientWpf.csproj
+```
+- From the `PollyTestClientWpf` directory:
```
Comment (martincostello) [inline PollyTestClientWpf/README.md:46]:
```suggestion
- From the `PollyTestClientWpf` directory:

```

Comment (martincostello) [inline PollyTestClientWpf/README.md:46]:
```suggestion
- From the `PollyTestClientWpf` directory:

```

File: PollyTestClientWpf/README.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+```none
+dotnet run
+```
+
+> [!IMPORTANT]
+> **Run the `PollyTestWebApi` as well**
```
Comment (martincostello) [inline PollyTestClientWpf/README.md:52]:
```suggestion
> **Run the `PollyTestWebApi` project as well**
```

File: PollyTestClientConsole/README.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+```none
+dotnet run --project PollyTestClientConsole/PollyTestClientConsole.csproj
+```
+- From the `PollyTestClientConsole` directory:
+```none
+dotnet run
+```
+
+> [!IMPORTANT]
+> **Run the `PollyTestWebApi` as well**
```
Comment (martincostello) [inline PollyTestClientConsole/README.md:44]:
```suggestion
> **Run the `PollyTestWebApi` project as well**
```

File: PollyTestWebApi/README.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+
+- It exposes three simple endpoints.
+- All of them echo back the received parameter (`{id}`) with some hard-coded prefixes.
+
+### `GET /api/Values/{id}`
+
+- It simply returns a string.
+- It is decorated with rate limiting.
+  - It allows 3 requests per 5 seconds.
+  - If the threshold is exceeded then it returns 429.
```
Comment (martincostello) [inline PollyTestWebApi/README.md:29]:
```suggestion
  - If the threshold is exceeded then it returns an HTTP 429 status code.
```

File: PollyTestWebApi/README.md [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+### `GET /api/NonThrottledGood/{id}`
+
+- It simply returns a string.
+- As its name suggests it is NOT decorated with rate limiting.
+
+### `GET /api/NonThrottledFaulting/{id}`
+
+- It waits **5 seconds** before returns a string.
+  - It emulates slow processing.
+- As its name suggests it is NOT decorated with rate limiting.
```
Comment (martincostello) [inline PollyTestWebApi/README.md:40]:
```suggestion
- As its name suggests it is **not** decorated with rate limiting.
```

File: PollyTestWebApi/README.md [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+
+## Structure
+
+- The `Program.cs` contains the majority of the codebase
+  - the rate limiting policy definition
+  - the controllers registration
+  - and it exposes the last two endpoint via Minimal API
+- The `Controllers/ValueController.cs`
```
Comment (martincostello) [inline PollyTestWebApi/README.md:48]: Let's make these relative links too.

File: PollyTestWebApi/README.md [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+  - contains the definition of the  `/api/Values/{id}` endpoint
+  - decorated with the rate limiting policy
+
+## How to run?
+
+- From the `PollySamples` directory:
```
Comment (martincostello) [inline PollyTestWebApi/README.md:54]:
```suggestion
- From the `PollySamples` directory:

```

**Comment by martincostello** [comment]
Nice work - just a bunch a various formatting and grammar suggestions.

**Comment by peter-csala** [comment]
@martincostello Is there anything else that needs to be done prior merging this PR?

**Comment by martincostello** [comment]
Just to patiently wait for me to have come back to a computer since 4pm yesterday and take a look 😉

**Review by martincostello — APPROVED** [review]

**Pull request merged**